application/src/main/java/org/thingsboard/server/service/query/DefaultEntityQueryService.java

```diff
@@ -208,9 +208,26 @@ public PageData<AlarmData> findAlarmDataByQuery(SecurityUser securityUser, Alarm
 
     @Override
     public long countAlarmsByQuery(SecurityUser securityUser, AlarmCountQuery query) {
+        if (query.getEntityFilter() != null) {
+            EntityDataQuery entityDataQuery = this.buildEntityDataQuery(query);
+            PageData<EntityData> entities = entityService.findEntityDataByQuery(securityUser.getTenantId(),
+                    securityUser.getCustomerId(), entityDataQuery);
+            if (entities.getTotalElements() > 0) {
+                List<EntityId> entityIds = entities.getData().stream().map(EntityData::getEntityId).toList();
+                return alarmService.countAlarmsByQuery(securityUser.getTenantId(), securityUser.getCustomerId(), query, entityIds);
+            } else {
+                return 0;
+            }
+        }
         return alarmService.countAlarmsByQuery(securityUser.getTenantId(), securityUser.getCustomerId(), query);
     }
 
+    private EntityDataQuery buildEntityDataQuery(AlarmCountQuery query) {
```

Comment:
Please also refactor `EntityDataQuery buildEntityDataQuery(AlarmDataQuery query)` . We don't need any entity fields or latest values for entities list, only ids